src/app.js

```diff
@@ -45,68 +48,82 @@ export default async function ({
+      }
+      // Hard coded wait if the items loaderProcess is still ongoing or if it errored (100 ms = 0,1 s)
+      return startCheck({checkInProcessItems: true, wait: 100, waitSinceLastOp});
     }
 
     if (prio) {
       logger.silly(`app/checkInProcess: Nothing found in process for PRIO -> checkInProcess for BULK `);
-      return checkInProcess(false);
+      return checkInProcess({prio: false, waitSinceLastOp});
     }
 
     logger.silly(`app/checkInProcess: Nothing found for BULK -> startCheck for importing without waiting`);
-    return startCheck(false);
+    return startCheck({checkInProcessItems: false, waitSinceLastOp});
   }
 
   // eslint-disable-next-line max-statements
-  async function checkItemImportingAndInQueue(prio = true) {
+  async function checkItemImportingAndInQueue({prio = true, waitSinceLastOp}) {
     const mongoOperator = prio ? mongoOperatorPrio : mongoOperatorBulk;
     const itemImportingHandler = prio ? prioItemImportingHandler : bulkItemImportingHandler;
     // Items in importer to be send to aleph-record-load-api
     // ImportJobStates: EMPTY, QUEUING, IN_QUEUE, PROCESSING, DONE, ERROR, ABORT
     // get here {<OPERATION>: IN_QUEUE}
 
-    if (await checkImportJobStateIMPORTING()) {
-      return startCheck();
+    // Next checks return true if they found and handled a queueItem, false, if they didn't find one
+
+    // queueItemState: IMPORTING, importJob: IMPORTING
+    if (await checkImportJobStateIMPORTING({waitSinceLastOp})) {
+      return startCheck({});
     }
 
-    if (await checkImportJobStateDONE()) {
-      return startCheck();
+    // queueItemState: IMPORTING, importJob: DONE
+    if (await checkImportJobStateDONE({waitSinceLastOp})) {
+      return startCheck({});
     }
 
-    if (await checkImportJobStateINQUEUE()) {
-      return startCheck();
+    // queueItemState: IMPORTING, importJob: IN_QUEUE
+    if (await checkImportJobStateINQUEUE({waitSinceLastOp})) {
+      return startCheck({});
     }
 
-    if (await checkQueueItemStateINQUEUE()) {
-      return startCheck();
+    // queueItemState: IN_QUEUE
+    if (await checkQueueItemStateINQUEUE({waitSinceLastOp})) {
+      return startCheck({});
     }
 
     if (prio) {
-      logger.silly(`app/checkItemImportingAndInQueue: Nothing found: PRIO -> checkItemImportingAndInQueue `);
-      return checkItemImportingAndInQueue(false);
+      logger.silly(`app/checkItemImportingAndInQueue: Nothing found: PRIO -> checkItemImportingAndInQueue for bulk`);
+      return checkItemImportingAndInQueue({prio: false, waitSinceLastOp});
     }
 
-    logger.silly(`app/checkItemImportingAndInQueue: Nothing found: BULK -> startCheck `);
-    return startCheck(true, 3000);
+    logger.silly(`app/checkItemImportingAndInQueue: Nothing found: BULK -> startCheck, waiting ${pollWaitTime} ms = ${pollWaitTime / 1000} s`);
+    return startCheck({checkInProcess: true, wait: pollWaitTime, waitSinceLastOp});
```

Comment:
This fixes hard coded waitTime of 3000 ms to waitTime configured in the environment variable POLL_WAIT_TIME